src/main/java/io/deepstream/DeepstreamClient.java

```diff
@@ -62,15 +62,31 @@ public ConnectionState getConnectionState() {
     }
 
     public String getUid() {
-        Long timestamp = new Date().getTime();
-        Double randomNumber = Math.random() * 10000000000000000D;
-        return timestamp + "-" + randomNumber.toString().replace(".", "");
+        if( config.get( "debug" ).equals( "true" ) ) {
```

Comment:
Yeah, having dynamically created ids are generally very hard to test.

I wouldn't use a debug statement like that though ( I would use it for extensive logging though ).

```
var matchMessage = function( actualMessage, expectedMessage ) {
	var expectedMessageCopy = expectedMessage;
	if( server.allMessages.length === 0 ) {
		return 'Server did not recieve any messages';
	}
	else if( expectedMessage.indexOf( '<UID>' ) === -1 && expectedMessage.indexOf( '<FIRST_SERVER_URL>' ) === -1 ) {
		return check( 'last received message', expectedMessage, convertChars( actualMessage ) );
	} else {
		expectedMessage = expectedMessage.replace( /\|/g, '\\|' );
		expectedMessage = expectedMessage.replace( '+', '\\+' );
		expectedMessage = expectedMessage.replace( '<UID>', '([^\\|]*)' );
		expectedMessage = expectedMessage.replace( '<FIRST_SERVER_URL>', 'localhost:' + config.testServerPort );

		var match = convertChars( actualMessage ).match( new RegExp( expectedMessage ) );
		if( match ) {
			uid = match[ 1 ];
			return;
		} else {
			return convertChars( actualMessage ) + ' did not match ' + expectedMessage;
		}
	}
};
```

Be good to not need to change the specs either ( other than the #scenario and bug fixes ) since they should be uniform.

That being said, you can leave this in as a //TODO and I can look at it once I get a bit of time!